.github/workflows/ci.yml

```
@@ -0,0 +1,29 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - main
+      - develop
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+
+permissions:
+  checks: write
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: Run tests
+        uses: ./.github/actions/run-tests
```

Since there's just one test job, consider inlining those actions' steps here.

We also don't need `nextest` here since it's a small package (i.e. parallelizing tests is not a big win) and we're not using JUnit output (the other reason I opted to use nextest)

```suggestion
        uses: actions-rs/toolchain@v1
        with:
          toolchain: nightly
          profile: minimal
      - name: Run tests
        run: cargo test --locked --no-fail-fast
```